src/layer/internal/mod.rs

```diff
@@ -743,6 +743,39 @@ impl Layer for InternalLayer {
         None
     }
 
+    fn id_object_is_node(&self, id: u64) -> Option<bool> {
+        if id == 0 {
+            return None;
+        }
+
+        let mut corrected_id = id - 1;
```

Comment:
The code starting here seems to be duplicated in some variation throughout this file. Is there a way to avoid this?

Reply:
Yes, it's very similar to the object lookup itself, the difference being that what we do on the inside is different.
To make it generic, I think I'd have to either write a macro, or have a function that takes a function to perform the inner part.

I propose we make an issue for it and do such a refactor later.